### PR TITLE
pusher: add tracing

### PR DESCRIPTION
**Pull request by acud**
adds a main trace to the pusher under which all the chunks in the snapshot will be visible on jaeger.

### CODE REVIEW

**Review by janos — APPROVED** [review]
Nice @acud. LGTM.

File: pkg/pusher/pusher.go [hunk 1]
```diff
@@ -79,6 +85,10 @@ LOOP:
 				break
 			}
 
+			if span == nil {
+				span, _, ctx = s.tracer.StartSpanFromContext(cctx, "pusher-sync-batch", s.logger)
```
Comment (pradovic) [inline pkg/pusher/pusher.go:89]: where will this new ctx be canceled?

**Review by pradovic — APPROVED** [review]
LGTM mod one question 👍

File: pkg/pusher/pusher.go (hunk 1 above)
Comment (acud) [reply pkg/pusher/pusher.go:89]: `ctx` will be on the first run the actual root context (`cctx`) which is being cancelled on `L64`, and once we reach this line then `ctx` becomes a child context of `cctx` and does not have its own cancel function, therefore it is not cancelled. if `cctx` is cancelled, then this context will be cancelled too. at least that is to the best of my understanding

Comment (pradovic) [reply pkg/pusher/pusher.go:89]: SGTM

**Pull request merged**